components/graph/git-node.js

```diff
@@ -24,12 +27,33 @@ var GitNodeViewModel = function(graph, sha1) {
   this.refs = ko.computed(function() {
     var rs = self.branchesAndLocalTags().concat(self.remoteTags());
     rs.sort(function(a, b) {
+      if (b.current()) return 1;
+      if (a.current()) return -1;
       if (a.isLocal && !b.isLocal) return -1;
       if (!a.isLocal && b.isLocal) return 1;
       return a.refName < b.refName ? -1 : 1;
     });
     return rs;
   });
+  // These are split up like this because branches and local tags can be found in the git log,
+  // whereas remote tags needs to be fetched with another command (which is much slower)
+  this.branches = ko.observableArray();
+  this.branchesToDisplay = ko.observableArray();
+  this.tags = ko.observableArray();
+  this.tagsToDisplay = ko.observableArray();
+  this.refs.subscribe((newValue) => {
+    if (newValue) {
+      this.branches(newValue.filter((r) => r.isBranch));
+      this.tags(newValue.filter((r) => r.isTag));
+      this.tagsToDisplay(this.tags.slice(0, maxTagsToDisplay));
+      this.branchesToDisplay(this.branches.slice(0, ungit.config.numRefsToShow - this.tagsToDisplay().length));
+    } else {
+      this.branches.removeAll();
+      this.tags.removeAll();
+      this.branchesToDisplay.removeAll();
+      this.tags.removeAll();
```

Comment:
Too late but I think this should be `this.tagsToDisplay.removeAll();`.

Reply:
thanks for the catch, will submit one soon if no one has